src/System.Management.Automation/DscSupport/CimDSCParser.cs

```diff
@@ -3261,14 +3261,15 @@ public static bool ImportCimKeywordsFromModule(PSModuleInfo module, string resou
                 //
                 try
                 {
-                    var dscResourceDirectories = Directory.GetDirectories(dscResourcesPath);
-                    foreach (var directory in dscResourceDirectories)
+                    foreach (var directory in Directory.EnumerateDirectories(dscResourcesPath))
                     {
-                        var schemaFiles = Directory.GetFiles(directory, "*.schema.mof", SearchOption.TopDirectoryOnly);
-                        if (schemaFiles.Length > 0)
+                        var schemaFiles = Directory.EnumerateFiles(directory, "*.schema.mof", SearchOption.TopDirectoryOnly);
+                        var tempSchemaFilepath = schemaFiles.FirstOrDefault();
```

Comment:
`FirstOrDefault` uses Linq so would be slower than just indexing if count >= 1

Reply:
We can't index as `schemaFiles` is an `IEnumerable<string>`:p

As the variable type was not apparent, I've made changes to conform with style guidance.